File: solutions/swift/kindergarten-garden/2/Sources/KindergartenGarden/KindergartenGarden.swift
enum Plant: Character {
  case clover = "C"
  case grass = "G"
  case radishes = "R"
  case violets = "V"
}

class Garden {
  let rows: [[Character]]
  init(_ diagram: String) { rows = diagram.split(separator: "\n").map(Array.init) }

  func plantsForChild(_ child: String) -> [Plant] {
    let index = Int(child.first!.asciiValue! - 65)
    return [
      Plant(rawValue: rows[0][2 * index])!,
      Plant(rawValue: rows[0][2 * index + 1])!,
      Plant(rawValue: rows[1][2 * index])!,
      Plant(rawValue: rows[1][2 * index + 1])!,
    ]
  }
}

File: solutions/ruby/kindergarten-garden/2/kindergarten_garden.rb
# frozen_string_literal: true

# class Garden
class Garden
  STUDENTS = %i[alice bob charlie david eve fred ginny harriet ileana joseph kincaid larry].freeze
  STUDENTS.each_with_index { |student, index| define_method(student) { plants_for(index) } }
  def initialize(diagram) = @rows = diagram.lines(chomp: true)

  private

  PLANTS = %i[clover grass radishes violets].map { [_1[0].upcase, _1] }.to_h
  def plants_for(index) = @rows.flat_map { |row| row[index * 2, 2].chars.map(&PLANTS) }
end
